indexer-app/src/cmr/indexer/data/elasticsearch.clj

```diff
@@ -161,6 +161,17 @@
         (info "Existing index set:" (pr-str existing-index-set))
         (info "New index set:" (pr-str expected-index-set))))))
 
+(defn delete-granule-index
+  "Delete an elastic index by name"
+  [context collection-concept-id index]
+  (index-set-svc/remove-deleted-granule-index context idx-set/index-set-id collection-concept-id)
+  (info (format "Deleting granule index %s from elastic" index))
+  (try
+    (esi/delete-index (context->conn context) index)
```

Comment:
what happens if deleting the index fails, do we want to remove the index set entry still?

Reply:
I'm thinking we would still want the index set entry removed. If a collection has been removed (or removal has been attempted) we are aiming to not write granules to a index for that collection. If the index set is not updated, future granules ingested to a new non-tombstone version of the collection will still point to the separate index and write there, so I would think we would prefer writing to small collections even if the index fails to delete